compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -485,14 +486,41 @@ private BlendedManifest createBlendedManifest(PackageContext rootPackageContext,
 
     private ResolutionOptions getResolutionOptions(PackageContext rootPackageContext,
                                                    CompilationOptions compilationOptions) {
+        boolean sticky = getSticky(rootPackageContext);
+        PackageLockingMode packageLockingMode;
+        SemanticVersion prevDistributionVersion = rootPackageContext.dependencyManifest().distributionVersion();
+        SemanticVersion currentDistributionVersion = SemanticVersion.from(RepoUtils.getBallerinaShortVersion());
+        if (!sticky && (null == prevDistributionVersion
+                || isMinorOrMajorVersionIncrement(prevDistributionVersion, currentDistributionVersion))) {
+            packageLockingMode = PackageLockingMode.SOFT;
+            DiagnosticInfo diagnosticInfo = new DiagnosticInfo(
+                    DiagnosticWarningCode.BUILT_WITH_OLDER_SL_UPDATE_DISTRIBUTION.diagnosticId(),
+                    "WARNING: Detected packages built with an older SL Update distribution. " +
```

Comment:
```suggestion
                    "Detected packages built with an older SL Update distribution. " +
```
`WARNING` will be printed by the `Diagnostic.toString()` method

Reply:
Let's improve this logic to print the warning only if the Dependencies.toml exists. i.e We don't need to warn if it is a fresh project.

Reply:
+1